common/autoware_universe_utils/src/math/trigonometry.cpp

```diff
@@ -72,4 +72,34 @@ std::pair<float, float> sin_and_cos(float radian)
   }
 }
 
+static const float atan2_p1 =
+  0.9997878412794807f * static_cast<float>(180) / autoware::universe_utils::pi;
+static const float atan2_p3 =
+  -0.3258083974640975f * static_cast<float>(180) / autoware::universe_utils::pi;
+static const float atan2_p5 =
+  0.1555786518463281f * static_cast<float>(180) / autoware::universe_utils::pi;
+static const float atan2_p7 =
+  -0.04432655554792128f * static_cast<float>(180) / autoware::universe_utils::pi;
+static const float atan2_DBL_EPSILON = 2.2204460492503131e-016f;
+
+float opencv_fastAtan2(float dy, float dx)
+{
+  float ax = std::abs(dx), ay = std::abs(dy);
+  float a, c, c2;
+  if (ax >= ay) {
+    c = ay / (ax + atan2_DBL_EPSILON);
+    c2 = c * c;
+    a = (((atan2_p7 * c2 + atan2_p5) * c2 + atan2_p3) * c2 + atan2_p1) * c;
+  } else {
+    c = ax / (ay + atan2_DBL_EPSILON);
+    c2 = c * c;
+    a = 90.f - (((atan2_p7 * c2 + atan2_p5) * c2 + atan2_p3) * c2 + atan2_p1) * c;
+  }
+  if (dx < 0) a = 180.f - a;
+  if (dy < 0) a = 360.f - a;
+
+  a = a * autoware::universe_utils::pi / 180.f;
+  return a;
+}
+
```

Comment:
Is license clear? no need the license?

Reply:
@yukkysaito @drwnz 
Thanks, could you help me check whether this is enough https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/62434faf959d7324e0d826f4e2b44ee84a3faa1d

Reply:
Please rename the license such that it is clear that it is for OpenCV. For example, 
* put it in a directory called `third_party_licenses` and name it `opencv-license.md`.
* then, comment where the license can be found in the doc comment of the function (link to original repo and path to the license copy in Autoware).
* finally, add a comment to pointcloud_preprocessor's README.md to explain where third-party licenses can be found (reference the directory created above)

Reply:
See [Apache Software Foundation's licensing how-to](https://infra.apache.org/licensing-howto.html)

Reply:
@mojomex Thanks for the advice!
I added them in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/b9a5c56436665dfbc79a9e6fde6e7e93ee3e97f8. 
However, the pre-commit modifies the layout of the license in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/b9a5c56436665dfbc79a9e6fde6e7e93ee3e97f8. Not sure about whether this influence :(